qustion.md

```diff
@@ -0,0 +1,28 @@
+Hi Professor,
+
+After the second individual session, I though through the language design and found out that it might be a matter of the user, that whether they are good at programming or not.
+
+For the capacitor discharging diagram, in which a capacitor and a resistor were connected in serives, I come up with two versions of syntax design. The first version is like this: ```CAP + RES``` and the second version is like this ```inSeries(Cap, Res)```. During the session, you said that these two design might cause confusion to my physics teacher who doesnt know computer very much.
+
+However, if the user of this DSL is some physics scientists who do know programming, I still feel that the first verison is quite a good idea.(using ```+```, ```|``` and ```()```denoting how circuits are connected)
+
+Making my physics teacher as the user, I generate the third version of my idea, I hope you could give me some comments:)
+
+I was inspired by SQLs, in which they have synatic keywords like ```From```, ```where```, or ```and```, so I design a syntax as shown below:
+
+```Connect Cap and Res In Series```
+
+similarly, if they are connected in parellel, it would be like:
+
+```Connect Cap and Res In Parellel```
+
+To add another conponent, it would be something like this:
+
+```Add LB to Cap in Series```,
+where ```LB``` denotes "light bulbs"
+
```

Comment:
This seems like a good design to start thinking about implementing. A few questions:
- How would someone describe connecting more than one thing (in series or in parallel)?
- What parts of a program are always the same (e.g., `Connect`) and what parts are "blanks", to be filled in by the programmer?